README.md

```diff
@@ -632,6 +617,29 @@ MarkdownModule.forRoot({
 }),
 ```
 
+
+### provide marked extensions
+
+there is an Angular token `MARKDOWN_EXTENSIONS` that you can provide with array of extensions to be used
+
+usage example for `app.config.ts` (but it's the same for `NgModule` or `Component` providers)
+```ts
+import { ApplicationConfig } from '@angular/core';
+import { gfmHeadingId } from 'marked-gfm-heading-id';
+import { MARKDOWN_EXTENSIONS } from 'ngx-markdown';
+
+export const appConfig: ApplicationConfig = {
+  providers: [
+    // other providers
+    {
+      provide: MARKDOWN_EXTENSIONS,
+      useValue: [gfmHeadingId()],
+    },
```

Comment:
For this injection token, one option that's slightly more aligned with the new `provide*` function pattern, is to use a `with*` function that you pass in as a second parameter to the `provide` function, which allows users to add additional options like this use case.  So, for example, a user could do this in their `app.config`

```ts
providers: [
  provideMarkdown({ ...markdownConfig }, withExtensions([gfmHeadingId()]),
]
```

The `withExtensions` function would set up the appropriate injection token for the `provideMarkdown` to use.  You can see an example of this in the article I linked (search `withColor`): https://www.angulararchitects.io/en/blog/patterns-for-custom-standalone-apis-in-angular/

Angular's new `provideRouter` function has several `with*` feature functions, if you want to use that for inspiration as well.  E.g. `withPreloading` https://angular.io/api/router/withPreloading

You could potentially do the same with several of the optional features that are embedded in the Markdown config, like moving Clipboard options into a `withClipboard` feature function.  But that may be a bit more of an overhaul than @jfcere  may be comfortable with?

Reply:
I think we should be consistent
Either to add the extensions to the configuration interface
Or to use with for everything
What do you think @jfcere

Reply:
The main difference with the injection token, though, is that it's not integrated into the config interface right now.  So it's already sort of its own thing, and could be converted to a `with` function.  But I agree, it would be nice to lean all the way into that pattern for optional features.

Reply:
> I think we should be consistent Either to add the extensions to the configuration interface Or to use with for everything What do you think @jfcere

I agree that it should be consistent, honestly I never add time to really dive into the standalone components architecture so I am not sure which approach is best.

I always favored having providers within the `markdownModuleConfig` so that it kinda regroup everything that is used by the ngx-markdown library instead of adding providers in the `providers` property of the module and ending up with a bunch of stuff you are not clearly sure it is used by what module.

So think that way, I do think that the `MARKDOWN_EXTENSIONS` should be an option within the `markdownModuleConfig` but this should be a comment for the previous PR #474. So I would be confortable with not having a bunch of `withSomething` functions for now.

Does it make sense to you guys?

Reply:
I agree
I will do it right now

Reply:
@jfcere I totally agree on having the self-contained config vs having a bunch of separate providers.  But I feel like, when using the new `provide*` function paradigm, the `with*` feature functions feel a bit more natural, than passing in a large config to the function.  So, for example, rather than having this

```ts
function markedOptionsFactory(anchorService: AnchorService): MarkedOptions {
  const renderer = new MarkedRenderer();

  // fix `href` for absolute link with fragments so that _copy-paste_ urls are correct
  renderer.link = (href: string, title: string, text: string) => {
    return MarkedRenderer.prototype.link.call(renderer, anchorService.normalizeExternalUrl(href), title, text) as string;
  };

  return { renderer };
}

provideMarkdown({
    markedOptions: {
      provide: MarkedOptions,
      useFactory: markedOptionsFactory,
      deps: [AnchorService],
    },
    clipboardOptions: {
      provide: ClipboardOptions,
      useValue: {
        buttonComponent: ClipboardButtonComponent,
      },
    },
    sanitize: SecurityContext.NONE,
  }),
```

you could have something like this

```ts
function createMarkedOptions: MarkedOptions {
  const anchorService: AnchorService = inject(AnchorService);
  const renderer = new MarkedRenderer();

  // fix `href` for absolute link with fragments so that _copy-paste_ urls are correct
  renderer.link = (href: string, title: string, text: string) => {
    return MarkedRenderer.prototype.link.call(renderer, anchorService.normalizeExternalUrl(href), title, text) as string;
  };

  return { renderer };
}

provideMarkdown(
  createMarkedOptions(),
  withClipboard({
      buttonComponent: ClipboardButtonComponent,
   }),
 ),
```

I don't have a strong preference, but the latter feels a bit more ergonomic and in line with the new direction the framework is going with their APIs.

Reply:
So taking the exemple...

```ts
provideMarkdown({
    markedOptions: {
      provide: MarkedOptions,
      useFactory: markedOptionsFactory,
      deps: [AnchorService],
    },
    markedExtensions {
      provide: MarkedExtensions,
      useValue: [gfmHeadingId()],
    },
    clipboardOptions: {
      provide: ClipboardOptions,
      useValue: { buttonComponent: ClipboardButtonComponent },
    },
    sanitize: SecurityContext.NONE,
  }),
```

You would end up with 4 `with*` functions as such?

```ts
provideMarkdown({
    withMarkedOptions(getMarkedOptions()),
    withMarkedExtensions([gfmHeadingId()]),
    withClipboardOptions({ buttonComponent: ClipboardButtonComponent }),
    withSanitize(SecurityContext.NONE),
  }),
```

Reply:
Well, I was thinking the core `MarkedOptions` would be the first parameter of the `provideMarkdown` function, since those are core functionality.  And then the other types of feature options would be encapsulated by `with` functions.  

e.g.
```ts
provideMarkdown(
  getMarkedOptions(),
  withMarkedExtensions([gfmHeadingId()]),
  withClipboardOptions({ buttonComponent: ClipboardButtonComponent }),
  withSanitize(SecurityContext.NONE),
),
```
It's just making a slight delineation between core function and additional features.  But the other way could work too.